tasks/main.yml

```diff
@@ -2,16 +2,58 @@
   timezone:
     name: "{{ koha_server_timezone }}"
 
-- name: Add Koha package repository signing key
+- name: Check to have {{ koha_domain }} domain in /etc/hosts
+  become: yes
+  lineinfile:
+    path: /etc/hosts
+    line: "127.0.1.1 {{ koha_domain }}"
+    insertafter: '127\.0\.0\.1 localhost'
+    state: present
+
+
+
+- name: Add Koha package community repository signing key
   apt_key:
     keyserver: "{{ koha_keyserver }}"
     id: "{{ koha_signing_key_id }}"
+  when: koha_signing_key_id is defined
+
+- name: Add Koha package community repository signing key by content
+  apt_key:
+    data: "{{ koha_signing_key_content }}"
+  when: koha_signing_key_content is defined
 
-- name: Add Koha package repository
+- name: Add Koha package community repository
   apt_repository:
     repo: "{{ koha_package_repository }}"
     state: present
-    filename: koha-packages
+    filename: koha-community
+
+- name: Add Koha package our repository signing key
+  apt_key:
+    keyserver: "{{ koha_extra_keyserver }}"
+    id: "{{ koha_extra_signing_key_id }}"
+  when: koha_extra_signing_key_id is defined and koha_extra_keyserver is defined
+
+- name: Add Koha package our repository signing key by content
+  apt_key:
+    data: "{{ koha_extra_signing_key_content }}"
+  when: koha_extra_signing_key_content is defined
+
+- name: Add Koha package our repository
+  apt_repository:
+    repo: "{{ koha_extra_package_repository }}"
+    state: present
+    filename: koha-kcminni
+
+- name: Configure pinning for koha-common
+  template:
+    src: pin-koha.j2
+    dest: /etc/apt/preferences.d/koha
+    owner: root
+    group: root
+    mode: u=rw,g=r
+  when: koha_extra_package_rep_domain is defined
```

Comment:
The name doesn't tell me clearly what this is about, I think this should be documented along with the other variables in README.md. Also here there is the "is defined" I mentioned earlier, e.g. https://github.com/geerlingguy/ansible-role-gitlab there is defaults/main.yml variables with just "" as the value so maybe we could do the same, but I also see it doing default(false) but that is just extra work in my opinion and the role user is unlikely to somehow undefine the variable. https://docs.ansible.com/ansible/latest/user_guide/playbooks_best_practices.html#keep-it-simple https://medium.com/opsops/is-defined-in-ansible-d490945611ae

Reply:
In a case variable will be not defined in file – script will fail if just to left "only" variable.
So I propose and did this in both-checks way, now it works properly in all cases:

```koha_signing_key_content is defined and koha_signing_key_content|length > 0```

(for example https://www.shellhacks.com/ansible-when-variable-is-defined-exists-empty-true/ https://stackoverflow.com/questions/36912726/how-to-test-that-a-registered-variable-is-not-empty/53362496 )